python/ray/serve/_private/deploy_utils.py

```diff
@@ -135,3 +138,14 @@ def deploy_args_to_deployment_info(
         autoscaling_policy=autoscaling_policy,
         is_driver_deployment=is_driver_deployment,
     )
+
+
+def get_app_code_version(app_config: ServeApplicationSchema):
+    encoded = json.dumps(
+        {
+            "import_path": app_config.import_path,
+            "runtime_env": app_config.runtime_env,
+        },
+        sort_keys=True,
+    ).encode("utf-8")
+    return hashlib.md5(encoded).hexdigest()
```

Comment:
There's already a `ReplicaVersion` struct that implements similar logic for the `user_config` -- could we try to unify this with that?

Reply:
Code version is passed to `DeploymentVersion`, which constructed in the deployment state manager after the deploy task finishes building the application, so it seems the simplest to hash the import path + runtime env into a code version when the config is first received. Maybe we can optimize this more in the future.